src/gui/tray/activitydata.cpp

```diff
@@ -82,10 +98,15 @@ OCC::Activity Activity::fromActivityJson(const QJsonObject &json, const AccountP
                 parameterJsonObject.value(QStringLiteral("id")).toString(),
                 parameterJsonObject.value(QStringLiteral("name")).toString(),
                 parameterJsonObject.contains(QStringLiteral("path")) ? parameterJsonObject.value(QStringLiteral("path")).toString() : QString(),
-                parameterJsonObject.contains(QStringLiteral("link")) ? QUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
+                parameterJsonObject.contains(QStringLiteral("link")) ? stringToUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
```

Comment:
You can make this part shorter and more efficient by always calling `stringToUrl(parameterJsonObject.value(QStringLiteral("link")).toString())` and then checking for `link.isEmpty()`. By default, the `toString()` will be empty if `parameterJsonObject.contains(QStringLiteral("link")) is false`.